core/src/main/scala/chisel3/Mem.scala

```diff
@@ -318,6 +318,9 @@ object SyncReadMem {
     if (compileOptions.declaredTypeMustBeUnbound) {
       requireIsChiselType(t, "memory type")
     }
+    if (ruw != Undefined) {
+      Builder.warning(s"specifying read-under-write behavior other than Undefined is deprecated 3.6 and removed in 5.0")
+    }
```

Comment:
Leaving this and also deprecating the `apply(Int, T, ReadUnderWrite)` methods would provide a compile-time warning.

Reply:
Not sure.. if we are planning on adding it back, I'm not sure its worth going through a round of removing the API then adding it back. Do you feel strongly?

Reply:
I don't feel strongly.

Reply:
@jackkoenig ?